client/src/components/Dashboard.js

```diff
@@ -3,69 +3,47 @@
-import ItemForm from './ItemForm';
-import { UserMenuContext } from '../App';
-import { dashboardReducer, initialState } from '../hooks/dashboardReducer';
+import { useFetch } from '../hooks/useFetch';
 
 export const MenuItemsContext = createContext();
 
 const Dashboard = () => {
-  const result = useContext(UserMenuContext);
-  const [menuItems, setMenuItems] = useState([]);
   const [anchorEl, setAnchorEl] = useState(null);
-  const [state, dispatch] = useReducer(dashboardReducer, initialState);
-  const { menuItemsPage, orderPage, reportPage } = state;
+  const { menuId } = useParams();
 
-  useEffect(() => {
-    fetch(`/items/${result.menu._id}`)
-      .then((res) => res.json())
-      .then((menuItems) => {
-        setMenuItems(menuItems);
-      });
-  }, [result.menu._id]);
-
-  const handleClick = (event) => {
-    setAnchorEl(event.currentTarget);
-  };
-
-  const handleClose = (e) => {
-    setAnchorEl(null);
-  };
+  const { data, setData, error } = useFetch(`/items/${menuId}`);
+  if (error) return <h1>{error}</h1>;
 
   return (
     <>
-      <Button aria-controls="simple-menu" aria-haspopup="true" onClick={handleClick} style={{ float: 'left' }}>
+      <Button
+        aria-controls="simple-menu"
+        aria-haspopup="true"
+        onClick={(e) => setAnchorEl(e.currentTarget)}
+        style={{ float: 'left' }}>
         Options
       </Button>
-      <Menu id="simple-menu" anchorEl={anchorEl} keepMounted open={Boolean(anchorEl)} onClose={handleClose}>
-        <MenuItem onClick={() => dispatch({ type: 'ORDER_PAGE' })} id="Order">
-          Order
+      <Menu id="simple-menu" anchorEl={anchorEl} keepMounted open={Boolean(anchorEl)} onClose={() => setAnchorEl(null)}>
+        <MenuItem onClick={() => setAnchorEl(null)} id="Order">
+          <Link to="order">Order</Link>
+        </MenuItem>
+        <MenuItem onClick={() => setAnchorEl(null)} id="Menu Options">
+          <Link to="items">Menu Options</Link>
         </MenuItem>
-        <MenuItem onClick={() => dispatch({ type: 'MENU_OPTIONS_PAGE' })} id="Menu Options">
-          Menu Options
+        <MenuItem onClick={() => setAnchorEl(null)} id="Reports">
+          <Link to="reports">Reports</Link>
         </MenuItem>
-        <MenuItem onClick={() => dispatch({ type: 'REPORT_PAGE' })} id="Reports">
-          Reports
+        <MenuItem onClick={() => setAnchorEl(null)} id="Log Off">
+          <Link to="/">Log Out</Link>
         </MenuItem>
       </Menu>
 
-      {orderPage ? (
-        <MenuItemsContext.Provider value={{ menuItems, setMenuItems }}>
-          <OrderContainer />
-        </MenuItemsContext.Provider>
-      ) : null}
-
-      {menuItemsPage ? (
-        <MenuItemsContext.Provider value={{ menuItems, setMenuItems, menuItemsPage }}>
-          <ItemForm />
-        </MenuItemsContext.Provider>
-      ) : null}
-
-      {reportPage ? <ReportsContainer /> : null}
+      <MenuItemsContext.Provider value={{ data, setData }}>
```

Comment:
I'm passing setData so when you add new items or make changes to the the items it will update the state which I'm sharing via context. You don't have to use context. You could pass props too.